diff -urN purestorage-1.19.0.orig/purestorage/purestorage.py purestorage-1.19.0/purestorage/purestorage.py
--- purestorage-1.19.0.orig/purestorage/purestorage.py	2021-03-14 19:47:30.647798338 -0400
+++ purestorage-1.19.0/purestorage/purestorage.py	2021-03-17 08:11:44.564541482 -0400
@@ -177,7 +177,13 @@
                 if response.cookies:
                     self._cookies.update(response.cookies)
                 else:
-                    self._cookies.clear()
+                    cookies_field = response.headers.get("Set-Cookie", "")
+                    if cookies_field:
+                        print("FlashArray SDK response.cookies empty. Trying to parse response.headers to get cookies")
+                        cookies = cookies_field.split("=", 1)
+                        self._cookies.update({cookies[0]:cookies[1]})
+                    else:
+                        self._cookies.clear()
                 content = response.json()
                 if isinstance(content, list):
                     content = ResponseList(content)
